docs/site/tutorials/core/10-advanced-recipes.md

```diff
@@ -0,0 +1,139 @@
+---
+lang: en
+title: 'Advanced Recipes'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/core-tutorial-part10.html
+---
+
+There are a few advanced usages of the LoopBack core modules that are not
+covered by the Greeter Extension and Greeter Application example.
+
+# Create your own decorator
+
+You can create your own decorator:
+
+1. Create a new decorator from scratch by using `DecoratorFactory` from
+   `@loopback/metadata`. See example from
+   [custom-inject-decorator.ts](https://github.com/strongloop/loopback-next/blob/master/examples/context/src/custom-inject-decorator.ts)
+
+2. Create a sugar decorator for an existing decorator.
+
+   ```ts
+   export function globalInterceptor(group?: string) {
+     bind({tags: [ContextTags.GLOBAL_INTERCEPTOR]});
+   }
+   ```
+
+# Create your own injector
+
+```ts
+export function env(name: string) {
+  return inject('', {resolve: () => process.env[name]});
+}
+```
+
+For a complete example, see
+https://github.com/strongloop/loopback-next/blob/master/examples/context/src/custom-inject-decorator.ts.
+
+# Class factory to allow parameterized decorations
+
```

Comment:
Maybe explain why it's needed - as decorations applied on a top-level class cannot be have references to variables.